ng-dev/utils/assertion-typings.d.ts

```diff
@@ -0,0 +1,17 @@
+type UnionToIntersection<U> = (U extends unknown ? (union: U) => void : never) extends (
+  intersection: infer I,
+) => void
+  ? I
+  : never;
+
+type AssertedType<A> = A extends AssertionFn<infer T> ? T : never;
+type AllAssertedTypes<A extends readonly AssertionFn<unknown>[]> = {
+  [K in keyof A]: AssertedType<A[K]>;
+};
+type TupleToUnion<T> = T[keyof T & number];
```

Comment:
It does not seem like this deals with a tuple (in combination with `AllAssertedTypes`). I think we should name it: `ExtractValuesAsUnionFromObject`. what do you think? `AllAssertedTypes` returns an object type like this IIUC:

```ts
type = {
   0: {github: GithubConfig},
   1: {release: ReleaseConfig},
}
```

Reply:
Yeah, we can do that, and correct I believe that is what the type looks like from `AllAssertedTypes`